lib/streamlit/elements/widgets/radio.py

```diff
@@ -243,6 +249,17 @@ def _radio(
                 "Radio index must be between 0 and length of options"
             )
 
+        def handle_captions(caption: Optional[str]) -> str:
+            if caption is None:
+                return ""
+            elif isinstance(caption, str):
+                return caption
+            else:
+                raise StreamlitAPIException(
+                    "Radio captions must be strings. Passed type: %s"
+                    % type(caption).__name__
+                )
```

Comment:
```suggestion
                raise StreamlitAPIException(
                    f"Radio captions must be strings. Passed type: {type(caption).__name__}"
                )
```

You can use f-strings instead of old-style % formatting here